airbyte-integrations/connectors/source-stripe/source_stripe/streams.py

```diff
@@ -88,150 +104,256 @@ def chunk_dates(self, start_date_ts: int) -> Iterable[Tuple[int, int]]:
+            )
+
+    @classmethod
+    def set_updated_timestamp(cls, records: Iterable[MutableMapping]) -> Iterable[MutableMapping]:
+        for record in records:
+            # need to fill in the `updated` timestamp for consistency
+            if "updated" not in record:
+                record["updated"] = record[cls.legacy_cursor_field]
+            yield record
 
-    def get_start_timestamp(self, stream_state) -> int:
-        start_point = self.start_date
-        if stream_state and self.cursor_field in stream_state:
-            start_point = max(start_point, stream_state[self.cursor_field])
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: Optional[List[str]] = None,
+        stream_slice: Optional[Mapping[str, Any]] = None,
+        stream_state: Optional[Mapping[str, Any]] = None,
+    ) -> Iterable[StreamData]:
+        if not stream_state:
+            # both full refresh and initial incremental sync should use usual endpoints
+            yield from self.set_updated_timestamp(
+                super().read_records(sync_mode, cursor_field=cursor_field, stream_slice=stream_slice, stream_state=stream_state)
+            )
+            return
+        yield from self.read_event_increments(cursor_field=cursor_field, stream_state=stream_state)
 
-        if start_point and self.lookback_window_days:
-            self.logger.info(f"Applying lookback window of {self.lookback_window_days} days to stream {self.name}")
-            start_point = int(pendulum.from_timestamp(start_point).subtract(days=abs(self.lookback_window_days)).timestamp())
 
-        return start_point
+class SlicedEventIncrementalStripeStream(EventIncrementalStripeStream, NormalIncrementalStripeStream, ABC):
+    """
+    This class combines both normal incremental sync and event based sync. It uses common endpoints for sliced data syncs in
+    the full refresh sync mode and initial incremental sync. For incremental syncs with a state, event based sync comes into action.
+    """
 
+    cursor_field = "updated"
 
-class Authorizations(IncrementalStripeStream):
+    def stream_slices(
+        self, sync_mode: SyncMode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        parent = EventIncrementalStripeStream if stream_state else NormalIncrementalStripeStream
+        return parent.stream_slices(self, sync_mode, cursor_field=cursor_field, stream_state=stream_state)
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: Optional[List[str]] = None,
+        stream_slice: Optional[Mapping[str, Any]] = None,
+        stream_state: Optional[Mapping[str, Any]] = None,
+    ) -> Iterable[StreamData]:
+        parent_read_records = EventIncrementalStripeStream.read_records if stream_state else NormalIncrementalStripeStream.read_records
```

Comment:
This means that the availability check will never query using `EventIncrementalStripeStream` and therefore will not be exhaustive for incremental syncs

Reply:
Yes, looks like you're right. Do you think we should modify availability strategy to accept stream state and pass it further?
We're passing state when performing a read operation, but for some reason we're not passing it when making a "test" read
![image](https://github.com/airbytehq/airbyte/assets/7703835/7a1a498c-7b59-480b-93ff-af050302ee7d)

Reply:
For the check, the Airbyte protocol does not support receiving a state so we won't have access to it here. Maybe we can wait for now but be very explicit that the availability check will not be exhaustive and why. In case there is an issue, we won't have to redo the investigation as to why the check does not work